pinot-controller/src/test/java/org/apache/pinot/controller/api/resources/PinotLogicalTableResourceTest.java

```diff
@@ -129,90 +133,191 @@ public void testCreateUpdateDeleteLogicalTables(String logicalTableName, List<St
     logicalTableConfig = getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNames, BROKER_TENANT);
     try {
-      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
       fail("Logical Table POST request should have failed");
     } catch (IOException e) {
       assertTrue(e.getMessage().contains("Reason: 'test_table_1' should be one of the existing tables"),
           e.getMessage());
     }
+  }
 
-    // Test valid broker tenant is provided
-    logicalTableConfig = getDummyLogicalTableConfig(LOGICAL_TABLE_NAME, physicalTableNamesWithType, "InvalidTenant");
+  @Test
+  public void testLogicalTableNameCannotSameAsPhysicalTableNameValidation()
+      throws IOException {
+    List<String> physicalTableNamesWithType = createHybridTables(List.of("test_table_1"));
+    LogicalTableConfig logicalTableConfig =
+        getDummyLogicalTableConfig("test_table_1", physicalTableNamesWithType, BROKER_TENANT);
     try {
-      ControllerTest.sendPostRequest(addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
       fail("Logical Table POST request should have failed");
     } catch (IOException e) {
-      assertTrue(e.getMessage().contains("Reason: 'InvalidTenant' should be one of the existing broker tenants"),
+      assertTrue(e.getMessage().contains("Table name: test_table_1 already exists"), e.getMessage());
+    }
+  }
+
+  @Test
+  public void testLogicalTableNameSuffixValidation()
+      throws IOException {
+    List<String> physicalTableNamesWithType = createHybridTables(List.of("test_table_1"));
+
+    // Test logical table name with _OFFLINE and _REALTIME is not allowed
+    LogicalTableConfig logicalTableConfig =
+        getDummyLogicalTableConfig("testLogicalTable_OFFLINE", physicalTableNamesWithType, BROKER_TENANT);
+    try {
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: 'tableName' should not end with _OFFLINE or _REALTIME"),
           e.getMessage());
     }
+
+    logicalTableConfig =
+        getDummyLogicalTableConfig("testLogicalTable_REALTIME", physicalTableNamesWithType, BROKER_TENANT);
+    try {
+      ControllerTest.sendPostRequest(_addLogicalTableUrl, logicalTableConfig.toSingleLineJsonString(), getHeaders());
+      fail("Logical Table POST request should have failed");
+    } catch (IOException e) {
+      assertTrue(e.getMessage().contains("Reason: 'tableName' should not end with _OFFLINE or _REALTIME"),
+          e.getMessage());
+    }
+  }
+
+  @DataProvider
```

Comment:
Can you check if there is already a table provider in `ClusterTest` ?

Reply:
NO. I did not find anything in `ClusterTest` or other places in the code.